### PR TITLE
chore: upgrade to rust 1.79

### PR DESCRIPTION
**Pull request by satyarohith**

### CODE REVIEW

File: ext/node/ops/crypto/mod.rs [hunk 1]
```diff
@@ -1458,15 +1429,6 @@ impl<'a> TryFrom<rsa::pkcs8::der::asn1::AnyRef<'a>>
...
         .unwrap_or(*SHA1_HASH_ALGORITHM);
 
-      let mask_gen_algorithm = decoder
-        .context_specific::<rsa::pkcs8::AlgorithmIdentifierRef>(
-          MASK_GEN_ALGORITHM_TAG,
-          pkcs8::der::TagMode::Explicit,
-        )?
-        .map(TryInto::try_into)
-        .transpose()?
-        .unwrap_or(*MGF1_SHA1_MASK_ALGORITHM);
```
Comment (littledivy) [inline ext/node/ops/crypto/mod.rs:1468]: `mask_gen_algorithm` is unused but we still want to do the parsing

**Review by littledivy — CHANGES_REQUESTED** [review]

File: cli/tools/registry/provenance.rs [hunk 2]
```diff
@@ -570,8 +570,8 @@ static DEFAULT_REKOR_URL: Lazy<String> = Lazy::new(|| {
 #[derive(Debug, Deserialize)]
 #[serde(rename_all = "camelCase")]
 pub struct LogEntry {
-  #[serde(rename = "logID")]
-  pub log_id: String,
+  // #[serde(rename = "logID")]
+  // pub log_id: String,
```
Comment (littledivy) [inline cli/tools/registry/provenance.rs:574]: I think we want to still keep this for deserialize validation.

File: cli/tools/registry/api.rs [hunk 3]
```diff
@@ -39,7 +39,7 @@ pub struct OidcTokenResponse {
 #[derive(serde::Deserialize)]
 #[serde(rename_all = "camelCase")]
 pub struct PublishingTaskError {
-  pub code: String,
+  // pub code: String,
```
Comment (littledivy) [inline cli/tools/registry/api.rs:42]: Ditto

File: cli/cdp.rs [hunk 4]
```diff
@@ -18,14 +17,6 @@ pub struct AwaitPromiseArgs {
   pub generate_preview: Option<bool>,
 }
 
-/// <https://chromedevtools.github.io/devtools-protocol/tot/Runtime/#method-awaitPromise>
-#[derive(Debug, Clone, Deserialize)]
-#[serde(rename_all = "camelCase")]
-pub struct AwaitPromiseResponse {
```
Comment (satyarohith) [inline cli/cdp.rs:24]: @crowlKats I hope removing these is fine.

**Review by littledivy — APPROVED** [review]

File: cli/cdp.rs (hunk 4 above)
Comment (crowlKats) [reply cli/cdp.rs:24]: Should be fine, originally implemented these for @bartlomieju , but I think it's fine to get rid

**Pull request merged**